src/snap/notificationObserver.ts

```diff
@@ -0,0 +1,33 @@
+import { DappeteerPage } from "../page";
+import { clickOnElement, openProfileDropdown } from "../helpers";
+import { NotificationList } from "./types";
+
+export const notificationObserver =
+  (page: DappeteerPage) => async (): Promise<NotificationList> => {
+    const notifications: NotificationList = [];
+
+    await page.bringToFront();
+    await openProfileDropdown(page);
+    await clickOnElement(page, "Notifications");
+
+    await page.exposeFunction("getItem", (message: string) => {
+      notifications.push({ message });
+    });
+
+    await page.evaluate(() => {
+      const observer = new MutationObserver((mutations) => {
+        for (const mutation of mutations) {
+          if (mutation.addedNodes.length) {
+            const element = mutation.addedNodes[0] as HTMLElement;
+            window.getItem(element.innerText);
+          }
+        }
+      });
+      observer.observe(document.querySelector(".notifications__container"), {
+        attributes: false,
+        childList: true,
+      });
+    });
+
+    return notifications;
```

Comment:
you should return to homepage before returning

Reply:
The first time we are calling this method we didn't invoke all the notifications yet. So it will be too early, we could add this as a part of the test. Also although it's what the user will want to do after the notifications, we will add too many things that the method will handle and it's not expected by the method name to come back to the home screen after getting all the notifications. I think it's better to create a helper method to come back to Home Screen

Reply:
I think the common principle of every dappeteer method is to leave everything as it was. This method leaves a dirty state of metamask and will break if a user tries to do something else after a method is done.